capstone-frontend/src/components/SearchBookModal.jsx

```diff
@@ -0,0 +1,238 @@
+import React, { Component } from 'react';
+import { Button, Form, Spinner, Modal, Col, Row } from 'react-bootstrap'
+
+class SearchBookModal extends Component {
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      fetchingBooks: false, // For Spinner
+      showModal: false,
+      typingTimeout: 0,
+      searchTerm: "",
+      searchResults: [],
+      displayBooks: false,
+      addedBooksIDs: [],
+      addedBooks: [],
+    }
+  }
+
+  getBooks = async (searchTerm) => {
+
+    this.setState({ fetchingBooks: true })
+
+    let searchResults;
+
+    if (searchTerm === "") {
+      searchResults = [];
+
+      this.setState({ searchResults, displayBooks: false, fetchingBooks: false })
+    }
+    else {
+      searchResults = await fetch(`/api/search?searchTerm=${searchTerm}&maxResults=${4}`)
+        .then(response => response.json())
+        .catch(err => alert(err));
+
+      if (typeof searchResults === "undefined") {
+        searchResults = [];
+      }
+
+      this.setState({ searchResults, displayBooks: true, fetchingBooks: false })
+    }
+  }
+
+  handleSearchTermChange = (event) => {
+
+    if (this.state.typingTimeout) {
+      clearTimeout(this.state.typingTimeout);
+    }
+
+    this.setState({
+      searchTerm: event.target.value,
+      typingTimeout: setTimeout(async () => {
+        await this.getBooks(this.state.searchTerm)
+      }, 500)
+    })
+  }
+
+  addBookToList = (book) => {
+
+    if (this.props.type === "club") {
+      this.setState(
+        {
+          addedBooksIDs: [book.id],
+          addedBooks: [book]
+        }
+      )
+    }
+    else {
+      this.state.addedBooksIDs.push(book.id);
+      this.state.addedBooks.push(book);
+
+      // Rerender
+      this.setState({ addedBooksIDs: this.state.addedBooksIDs, addedBooks: this.state.addedBooks })
+    }
+  }
+
+  removeBookFromList = (book) => {
+
+    const indexId = this.state.addedBooksIDs.indexOf(book.id);
+    this.state.addedBooksIDs.splice(indexId, 1);
+
+    const indexBook = this.state.addedBooks.indexOf(book);
+    this.state.addedBooks.splice(indexBook, 1)
+
+    // Rerender
+    this.setState({ addedBooksIDs: this.state.addedBooksIDs, addedBooks: this.state.addedBooks })
+  }
+
+  handleSubmit = async () => {
+
+    if (this.state.addedBooksIDs.length !== 0) {
+
+      await Promise.all(this.state.addedBooksIDs.map(async bookId => {
+
+        var updateJson;
+
+        if (this.props.type === "club") {
+          updateJson = {
+            "id": this.props.objectId,
+            "add_gbookID": bookId,
```

Comment:
This should read "gbookID" instead of "add_gbookID." When this modal is used for clubs, it will be setting the value of gbookID to be bookId, not appending this value to a list. Clubs do not have a list of books, just the current gbookID.

Reply:
ok